sys/stdio_nimble/stdio_nimble.c

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright (C) 2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys
+ * @{
+ *
+ * @file
+ * @brief       STDIO over NimBLE implementation
+ *
+ *
+ * @author      Hendrik van Essen <hendrik.ve@fu-berlin.de>
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include "nimble_riot.h"
+#include "nimble/nimble_port.h"
+#include "net/bluetil/ad.h"
+
+#include "host/ble_hs.h"
+#include "host/util/util.h"
+#include "host/ble_gatt.h"
+#include "services/gap/ble_svc_gap.h"
+#include "services/gatt/ble_svc_gatt.h"
+
+#if IS_USED(MODULE_STDIO_NIMBLE_DEBUG)
+#include <stdarg.h>
+#include "stdio_uart.h"
+#include "periph/uart.h"
+#endif /* IS_USED(MODULE_STDIO_NIMBLE_DEBUG) */
+
+#if IS_USED(MODULE_VFS)
+#include "vfs.h"
+#endif
+
+#include "tsrb.h"
+#include "isrpipe.h"
+#include "stdio_nimble.h"
+
+/* TODO: Why is 261 the absolute maximum before it stops working? */
+#define MTU_MAX_NIMBLE 261 //BLE_ATT_MTU_MAX
+
+/*
+ * PDU = LL Header (2 bytes)
+ *     + L2CAP Header (4 bytes)
+ *     + Op Code (1 byte)
+ *     + Attribute Handler (2 bytes)
+ *     + MIC (Optional, 4 bytes)
+ *     = 13 bytes
+ * */
+#define PDU_HEADERS_SIZE 13
+#define MAX_MTU_PAYLOAD (MTU_MAX_NIMBLE - PDU_HEADERS_SIZE)
+#define CALLOUT_TICKS    1
```

Comment:
Maybe add a comment that this ms

Reply:
How do we know it is ms? The documentation I found just states `@param[in]   ticks   number of ticks to wait before posting an event`. I saw some conversion functions for ms=>tick and tick=>ms, but I'm not sure whether we can say that 1 tick equals 1 ms.

Reply:
Nimble runs on ZTIMER_MSEC, everything is ms.